src/server/table.h

```diff
@@ -142,4 +138,11 @@ struct DbTable : boost::intrusive_ref_counter<DbTable, boost::thread_unsafe_coun
 // the snapshot process. We copy the pointers in StartSnapshotInShard function.
 using DbTableArray = std::vector<boost::intrusive_ptr<DbTable>>;
 
+// Before acquiring and releasing keys, one must "normalize" them via GetLockTag().
+LockTag GetLockTag(std::string_view key);
+
+inline LockFp GetLockFp(std::string_view key) {
+  return GetLockTag(key).Fingerprint();
+}
```

Comment:
I see, so you didn't like my ideas... 😿

Reply:
The type is never constructed differently except by a utility function in a totally different file

```c++
struct LockFp {
  LockFp(string_view key) : tag {IsCLuster() ? HashTag(key) : key} {
  }

 private:
  const string_view tag;
};
```

and we don't need `GetLockFp` or `GetLockTag` at all, we can have `LockTag{key}.Fingerprint()`, so there is one way of doing this, not utilities straying somewhere around

Reply:
Also transaction.h already includes table.h, so I guess we can move the whole type there instead of common.h

Reply:
conn_context does not include table.h